### PR TITLE
Fixes MoMMIs nullspacing their hat on death (fixes #8142)

### LINKED ISSUE #8142: Killed MoMMIs delete their items
Killed a MoMMI and the plastic bag that was on it's head with items I wanted got deleted once it blew up.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: code/modules/mob/living/silicon/mommi/inventory.dm [hunk 1]
```diff
@@ -166,17 +166,16 @@
...
-		// Put the hat in the MoMMI's claw
-		put_in_active_hand(head_state)
-
 		// Remove the head_state icon from the client's screen
 		if (client)
 			client.screen -= head_state
 
-		// Delete the hat from the head
+		head_state.forceMove(loc)
+		head_state.dropped(src)
```
Comment (PJB3005) [inline code/modules/mob/living/silicon/mommi/inventory.dm]: Doesn't `drop_item` do all of this?

Comment (Shadowmech88) [reply code/modules/mob/living/silicon/mommi/inventory.dm]: I remember hearing that MoMMIs use some weird bastardized inventory slots that don't behave quite like regular mob inventory slots.

Comment (PJB3005) [reply code/modules/mob/living/silicon/mommi/inventory.dm]: Even then I think having these 4 lines that should be a generic drop proc is a bit bad for maintainability.

**Review by PJB3005 — CHANGES_REQUESTED** [review]
See my comment

File: code/modules/mob/living/silicon/mommi/inventory.dm (hunk 1 above)
Comment (9600bauds) [reply code/modules/mob/living/silicon/mommi/inventory.dm]: MoMMI inventory code is ASS
ALL
OF
IT

I'm serious, check the rest of the file, drop_item doesn't even support anything but the held item

Comment (PJB3005) [reply code/modules/mob/living/silicon/mommi/inventory.dm]: Doesn't mean this maintainability mess is acceptable.

**Comment by 9600bauds** [comment]
God, I fucking hate MoMMI inventory code, as if it wasn't awful enough I never play silicon so I'm not even sure how silicon inventories are supposed to WORK

I hope I never have to touch this file again

File: code/modules/mob/living/silicon/mommi/death.dm [hunk 2]
```diff
@@ -31,32 +29,3 @@
 
 	dead_mob_list -= src
 	qdel(src)
-
-/mob/living/silicon/robot/mommi/death(gibbed)
```
Comment (9600bauds) [inline code/modules/mob/living/silicon/mommi/death.dm:35]: this was literally a 1:1 copypaste of /mob/living/silicon/robot/death(), and it STILL called the parent anyways so it did everything twice

File: code/modules/mob/living/silicon/mommi/inventory.dm [hunk 3]
```diff
@@ -62,23 +62,17 @@
 	update_items()
 	return 1
 
-//Attemps to remove an object on a mob.  Will not move it to another area or such, just removes from the mob.
-/mob/living/silicon/robot/mommi/remove_from_mob(var/obj/O)
```
Comment (9600bauds) [inline code/modules/mob/living/silicon/mommi/inventory.dm:66]: this was also a textual copypaste of /mob/proc/remove_from_mob(), for seemingly no reason, which obviously got outdated with the years

File: code/modules/mob/living/silicon/mommi/inventory.dm [hunk 4]
```diff
@@ -62,23 +62,17 @@
...
-	src.u_equip(O,0)
-	if (src.client)
-		src.client.screen -= O
-	O.reset_plane_and_layer()
-	O.screen_loc = null
-	return 1
-
 /mob/living/silicon/robot/mommi/u_equip(W as obj)
```
Comment (9600bauds) [inline code/modules/mob/living/silicon/mommi/inventory.dm:65]: this proc just wasn't used at all, just hackish copypaste all over the file to try to replace it

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**